GameServer/Logic/Maps/SummonersRift.cs

```diff
@@ -349,37 +419,91 @@ public override bool Spawn()
                 MinionSpawnPosition.SPAWN_RED_BOT,
                 MinionSpawnPosition.SPAWN_RED_MID,
             };
-
-            if (_waveNumber < 3)
+            foreach (var pos in positions)
             {
-                for (var i = 0; i < positions.Count; ++i)
+                uint inhibitorId = 0;
+                var waypoints = new List<Vector2>();
+                switch (pos)
                 {
-                    Minion m = new Minion(MinionSpawnType.MINION_TYPE_MELEE, positions[i], _laneWaypoints[i]);
-                    AddObject(m);
+                    case MinionSpawnPosition.SPAWN_BLUE_BOT:
+                        waypoints = _blueBotWaypoints;
+                        inhibitorId = 0xff26ac0f;
+                        break;
+                    case MinionSpawnPosition.SPAWN_BLUE_MID:
+                        waypoints = _blueMidWaypoints;
+                        inhibitorId = 0xffff8f1f;
+                        break;
+                    case MinionSpawnPosition.SPAWN_BLUE_TOP:
+                        waypoints = _blueTopWaypoints;
+                        inhibitorId = 0xff6793d0;
+                        break;
+                    case MinionSpawnPosition.SPAWN_RED_BOT:
+                        waypoints = _redBotWaypoints;
+                        inhibitorId = 0xff9303e1;
+                        break;
+                    case MinionSpawnPosition.SPAWN_RED_MID:
+                        waypoints = _redMidWaypoints;
+                        inhibitorId = 0xff4a20f1;
+                        break;
+                    case MinionSpawnPosition.SPAWN_RED_TOP:
+                        waypoints = _redTopWaypoints;
+                        inhibitorId = 0xffd23c3e;
+                        break;
                 }
-                return false;
-            }
+                var inhibitorIsDead = (GetInhibitorById(inhibitorId).getState() == InhibitorState.Dead);
 
-            if (_waveNumber == 3)
-            {
-                for (var i = 0; i < positions.Count; ++i)
+                if (_gameTime > 35 * 60 * 1000)
                 {
-                    Minion m = new Minion(MinionSpawnType.MINION_TYPE_CANNON, positions[i], _laneWaypoints[i]);
-                    AddObject(m);
+                    var list = inhibitorIsDead ? _superCannonMinionWave : _cannonMinionWave;
+                    SpawnMinion(list, _minionNumber, pos, waypoints);
                 }
-                return false;
-            }
+                else if (_gameTime > 20 * 60 * 1000)
+                {
+                    var list = inhibitorIsDead ?
```

Comment:
This has condition set has to be expanded over multiple lines, difficult to interpret like this as it's "compressed". Chaining the conditional operator is almost always a bad idea.